lib/Doctrine/ruleset.xml

```diff
@@ -30,4 +30,151 @@
             <property name="ignoreNewlines" value="true"/>
         </properties>
     </rule>
+
+    <!-- Forbid using absolute class name references (except global ones) -->
+    <rule ref="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly">
+        <properties>
+            <property name="allowFullyQualifiedGlobalClasses" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedGlobalFunctions" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedGlobalConstants" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedNameForCollidingClasses" type="boolean" value="true"/>
+        </properties>
+    </rule>
+
+    <!-- Forbid empty lines around type declarations -->
+    <rule ref="SlevomatCodingStandard.Types.EmptyLinesAroundTypeBraces">
+        <properties>
+            <property name="linesCountAfterOpeningBrace" value="0"/>
+            <property name="linesCountBeforeClosingBrace" value="0"/>
+        </properties>
+    </rule>
+
+    <!-- Require space around colon in return types -->
+    <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing">
+        <properties>
+            <property name="spacesCountBeforeColon" value="1"/>
+        </properties>
+    </rule>
+
+    <!-- Require types to be written as natively if possible;
+         require iterable types to specify phpDoc with their content;
+         forbid useless/duplicated information in phpDoc -->
+    <rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration">
+        <properties>
+            <property name="enableEachParameterAndReturnInspection" value="true"/>
+            <property name="traversableTypeHints" type="array" value="Doctrine\Common\Collections\Collection"/>
+            <property
+                name="usefulAnnotations"
+                type="array"
+                value="
+                    @dataProvider,
+                    @deprecated,
+                    @expectedException,
+                    @expectedExceptionMessage,
+                    @expectedExceptionMessageRegExp,
+                    @expectedExceptionCode,
+                    @expectedDeprecation,
+                    @group,
+                    @internal,
+                    @link,
+                    @see,
+                    @since,
+                    @throws
+                "
+            />
+        </properties>
+    </rule>
+
+    <!-- Require using Throwable instead of Exception -->
+    <rule ref="SlevomatCodingStandard.Exceptions.ReferenceThrowableOnly"/>
+
+    <!-- Require presence of declare(strict_types=1) -->
+    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
+        <properties>
+            <property
+                name="newlinesCountBetweenOpenTagAndDeclare"
+                value="2"
```

Comment:
Personally, I find this formatting better:
```php
<?php declare(strict_types=1);
```
It kind of makes the `declare` part of the file header.

Reply:
Let's stick to the (upcoming) PSR-12 as much as possible please. See https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#3-declare-statements-namespace-and-import-statements

> It kind of makes the `declare` part of the file header.

Which is exactly the problem. `declare` is a statement like many others and `strict_types` may not be the only statement:

```
<?php declare(strict_types=1);

declare(ticks = 1);
```

Reply:
Agreed with @alcaeus, declare could be anywhere in the file, it's only strict_types that must be first statement. Also in future it's possible there will be more similar declare statements.
Personally I don't see a reason to follow PSR-12, it's not a standard, is unfinished and Doctrine is not part of FIG group.
Btw we already use strict types in ORM (doctrine/doctrine2#6524) and format was discussed by @Ocramius.

Reply:
> Personally I don't see a reason to follow PSR-12, it's not a standard, is unfinished

Correct - it's currently still in a draft phase.

> and Doctrine is not part of FIG group.

Which means we are not involved in the creation and adoption of new PSRs. By the same logic, we might as well drop PSR-2, start using tabs and roll our own autoloading standard because we dislike PSR-0 and PSR-4 and we're not part of PHP-FIG.

Reply:
What I meant is that Doctrine is not part of FIG and is then not _obliged_ to follow PSR. On the other hand, it's not reason to not follow PSRs that are compatible with Doctrine's vision, like currently PSR-2.

Reply:
> Let's stick to the (upcoming) PSR-12 as much as possible please.

Didn't know it's a thing.